umap/static/umap/js/umap.controls.js

```diff
@@ -771,13 +807,12 @@ L.U.Map.include({
       if (!found)
         this.ui.alert({ content: L._('No results for these facets'), level: 'info' })
     }
-
-    const fields = keys.map((current) => [
-      `facets.${current}`,
+    const fields = keys.map((key) => [
+      `facets.${key}`,
       {
-        handler: 'FacetSearch',
-        choices: knownValues[current],
-        label: this.getFacetKeys()[current],
+        handler: ["date", "datetime-local", "number"].includes(facetCriteria[key]["inputType"]) ? 'FacetSearchMinMax' : 'FacetSearchChoices',
```

Comment:
I'd keep the handler choice out of the object for readability.

Reply:
I moved the code out of the object. There are now three variables in the mapping function, which are used in the object definition. Is that what you meant?